instrumentation/google.golang.org/grpc/otelgrpc/metadata_supplier_bench_test.go

```diff
@@ -0,0 +1,31 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelgrpc
+
+import (
+	"testing"
+
+	"go.opentelemetry.io/otel"
+)
+
+func BenchmarkMetadataSupplier(b *testing.B) {
+	ctx := b.Context()
+	propagator := otel.GetTextMapPropagator()
+
+	b.Run("extract", func(b *testing.B) {
+		b.ReportAllocs()
+		b.ResetTimer()
+		for range b.N {
```

Comment:
```suggestion
		for b.Loop() {
```

Adopting this method can reduce the idle running issues that may arise from compiler optimization. 

Relevant information: https://go.dev/blog/testing-b-loop

For other similar issues, please handle them together as well.